### PR TITLE
Enable nullable reference types

### PR DESCRIPTION
**Pull request by deepumi**
Enable C# 8.0 Nullable Reference Types for the following projects
1)  Ardalis.Specification
2)  Ardalis.Specification.IntegrationTests
3)  Ardalis.Specification.UnitTests

Locally I tested both unit test and integration test and looks like all test passed successfully!

image

@ardalis  Please take a look.

### CODE REVIEW

File: src/Ardalis.Specification/SpecificationEvaluator.cs [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-using System.Linq;
+﻿using System.Linq;
 using System.Threading.Tasks;
 
 namespace Ardalis.Specification
 {
     public class SpecificationEvaluator<T> where T : class
     {
-        public static async Task<IQueryable<T>> GetQuery(IQueryable<T> inputQuery, ISpecification<T> specification)
+        public static Task<IQueryable<T>> GetQuery(IQueryable<T> inputQuery, ISpecification<T> specification)
```
Comment (ardalis) [inline src/Ardalis.Specification/SpecificationEvaluator.cs:8]: Why did you remove async here?

File: src/Ardalis.Specification/SpecificationEvaluator.cs [hunk 2]
```diff
@@ -47,7 +46,7 @@ public static async Task<IQueryable<T>> GetQuery(IQueryable<T> inputQuery, ISpec
                 query = query.Skip(specification.Skip)
                              .Take(specification.Take);
             }
-            return query;
+            return Task.FromResult(query);
```
Comment (ardalis) [inline src/Ardalis.Specification/SpecificationEvaluator.cs:49]: Did you modify this due to a compiler warning, or ?

File: src/Ardalis.Specification/SpecificationEvaluator.cs (hunk 1 above)
Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:8]: @ardalis 
Good question, the method does not have any asynchronous operation like network calls and hence not required to add a 'async', However, if you want to consume on the other side we can still call the method like 'await GetQuery()' as it support Task<T>. Visual studio was showing warning that's why I have to make this change..

File: src/Ardalis.Specification/SpecificationEvaluator.cs (hunk 2 above)
Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: @ardalis Yes, that's correct. Since the method does not do any actual async operation, call and be still synchronous operation.

Comment (ardalis) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: True. In your opinion is there value in returning a task at all, instead of just returning `IQueryable<T>` ?

Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: Well, bases on the scenario below, yes its recommended. 

In REST API call, has some async calls to get a network or database request and this case all other method making to async / await is beneficial since it won't make the incoming thread to wait to process a synchronous call.  And making the whole request pipeline is async/await is recommended.  (https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md)

If we don't require any asyc/await call for the method we can simply return IQueryable<T>. Since this library using some places and I don't recommend to make a breaking change to async to sync.

Comment (fiseni) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: This method originally was synchronous. The method should do only in-memory operations, it just evaluates different expressions and translates into iqueryable. Any async calls to external libraries within the evaluator might affect overall performance (specs should be created on the fly). Externall calls should be responsibility of the repository.

Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: That's correct, adding a "async" will generate extra code on the fly. Please check both links with 'async' and with 'no async'

With Async = https://sharplab.io/#v2:D4AQTAjAsAUCAMACEECsBuWsQGZlkQGFEBvWRC5PEAFkQFkAKASlPMoF92LvEAHAE4BLAG4BDAC4BTZBABsyABzIFACSkAbDQHsA6toEaAJgEEAzgE8AdgGMA+gFEAHhIFiWZGJW+JeFEACcKgB0AGIC2gC2AEpSZgCuGhKMAETqWtqIAO4GxinM6JS8XDAcQA==

With out Async = https://sharplab.io/#v2:D4AQTAjAsAUCAMACEECsBuW55bAZkQG9ZFTkwwiSyaUBOACgCJAeDcBB9pgSkxhoF9qpQYgAOAJwCWANwCGAFwCmyCADZkagBIKANtoD2AdT1jtAEwCCAZwCeAOwDGDTsV40yw0iADs6gHQAxMT0AWwAlBUsAV205Zi1dPUQAd2MzLnR3V1IBGD4gA==

My proposal was to get rid of async usage, so we don't want to create any unnecessary allocations..  And I really don't see much values to make this method as async.. since its consuming other projects this would be a breaking change as I mentioned above.

Comment (ardalis) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: Right I'm thinking it should just be synchronous and return IQueryable which eliminates all of the async/Task crud from it (and as noted it doesn't actually do any async work).

Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: Awesome... If you want I can take care of it as a new PR once this gets approved or merged!

Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: Changing Async to Sync will eventually affect following files.. 

image

Comment (ardalis) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: If you don't mind, go ahead and make the change to sync. We don't need any of that to be async at this point - it was changed while trying to get GroupBy working with EF Core 3.x.

Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs:49]: Sure no problem... I will commit the changes now

Comment (deepumi) [reply src/Ardalis.Specification/SpecificationEvaluator.cs]: @ardalis, I just committed my changes. Please review https://github.com/ardalis/Specification/pull/21/commits/9bdca6be7a61da1df540cad521a5cf281dfa78db
